Magento_CatalogSearch/layout/default.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd"
+>
+    <body>
+        <referenceBlock name="catalog-search-advanced-link">
```

Comment:
IMO it will be way easier in terms of future maintenance to move all of these class updates to `Magento_Theme/layout/default.xml` and add some comment to mark this as a footer stuff.